test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs

```diff
@@ -27,6 +27,139 @@ public class OtlpLogExporterTests : Http2UnencryptedSupportTests
 {
     private static readonly SdkLimitOptions DefaultSdkLimitOptions = new();
 
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void ResolutionOrderTest(bool requestLoggerProviderDirectly)
```

Comment:
@Yun-Ting \ @CodeBlanch  - Could we add a short explanation on why we are adding this test?

Reply:
@vishweshbankwar I removed the test from here since it really didn't have anything to do with OtlpExporter and replaced it with one in `OpenTelemetry.Tests` called `CircularReferenceTest` which should be more self-explanatory. LMK!

Reply:
Diff for easy reviewing: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5103/commits/e208fbcf8ece49f4f96fc2df58f334c21cc4739b